app/dao/notifications_dao.py

```diff
@@ -138,6 +148,32 @@ def update_notification_status_by_id(
     if is_feature_enabled(FeatureFlag.NOTIFICATION_FAILURE_REASON_ENABLED) and status_reason:
         notification.status_reason = status_reason
 
+    # prevents sent -> sending
+    if (notification.status == NOTIFICATION_SENT) and (status == NOTIFICATION_SENDING):
+        current_app.logger.info(
+            'warning: attempt was made to transition notification id %s from %s to %s',
+            notification_id,
+            notification.status,
+            status
+        )
```

Comment:
@k-macmillan Do we need to log this?  It seems like something that will happen frequently.